public/pages/workflow_detail/component_details/input_fields/select_field.tsx

```diff
@@ -44,17 +32,51 @@ interface SelectFieldProps {
  * options.
  */
 export function SelectField(props: SelectFieldProps) {
-  const options = existingIndices;
+  // Redux store state
+  // Initial store is fetched when loading base <DetectorDetail /> page. We don't
+  // re-fetch here as it could overload client-side if user clicks back and forth /
+  // keeps re-rendering this component (and subsequently re-fetching data) as they're building flows
+  const models = useSelector((state: AppState) => state.models.models);
+
+  // Options state
+  const [options, setOptions] = useState<string[]>([]);
+
+  // Populate options depending on the select type
+  useEffect(() => {
+    if (props.field.selectType === 'model' && models) {
+      setOptions(Object.keys(models));
+    }
+  }, [models]);
```

Comment:
Do we need error handling here?

Reply:
There is no chance for NPEs as `models` is initialized to an empty map, `options` is initialized to an empty arr, and `selectType` can be null/undefined, in which case the if block will fail. If there was any errors, it will be logged when fetching models, and will just show as an empty dropdown list.

We will have any errors in redux store bubbled up via toasts or static callouts at the top of the pages; there is no UX for that for now.